### DISCUSSION TITLE
The expansion time of the osd node is too long

### DISCUSSION THREAD

**Discussion by fengjiankui121** [rook/rook#10545] [General]
When rook expands the osd node, since the new node takes more time to execute the ceph-vlume batch, the existing node will complete the provision faster, so the new node will enter the queue at the latest to wait for the creation of the osd deployment. If there are already many nodes in the cluster, it will take a lot of time. In order to save the expansion time, is it possible to interrupt the current comparison of the osd deployment of the existing nodes after the provisioning of the new node is completed, and immediately create the deploy of the osd of the new node? , After the osd of the new node is created, compare whether the osd deployment of the previous node has changed.

**Comment by travisn** [comment]
The new deployments are expected to have higher priority for creation before updating the existing OSDs, but sounds like that is not the observed behavior. @BlaineEXE Thoughts?

**Reply by BlaineEXE** [reply]
This sounds like a byproduct of the `ceph-volume` operations used to prepare new disks taking longer than it takes to update existing OSDs. But contrary to what seems to be stated, the newly provisioned OSDs are not pushed to the back of the queue. Once an OSD is done being provisioned, it is given the next highest priority after the current operation finishes.

**Reply by fengjiankui121** [reply]
How is “the next highest priority” guaranteed? @BlaineEXE

**Reply by BlaineEXE** [reply]
The function `updateAndCreateOSDsLoop()` codifies it in Rook. If there is no new OSD PVC or node provisioned within 100 milliseconds, Rook will update some OSDs. Once it's done, it will return to the loop.

**Reply by fengjiankui121** [reply]
Does updateTicker.C need to be changed to time.After(100 * time.Millisecond)? Otherwise, I think "the next highest priority" should not be guaranteed by the mechanism you mentioned. @BlaineEXE

**Reply by BlaineEXE** [reply]
I don't believe so. The ticker gets reset as noted in the comment associated with this code block.

https://github.com/BlaineEXE/rook/blob/0d8c581aa9fe8ae9d4d4b8f31ae5c97e44e8b1fa/pkg/operator/ceph/cluster/osd/status.go#L237-L242
